npm_and_yarn/lib/dependabot/npm_and_yarn/update_checker/version_resolver.rb

```diff
@@ -436,6 +450,15 @@ def run_npm_checker(path:, version:)
           end
         end
 
+        def run_npm7_checker(version:)
+          SharedHelpers.run_shell_command(
+            "npm install #{dependency.name}@#{version} --package-lock-only --ignore-scripts"
```

Comment:
Wondering if `{dependency.name}@#{version}` works for git dependencies, where we used to do this:

```js
  const source = (reqs.find((req) => req.source) || {}).source;

  if (source && source.type === "git") {
    return [`${depName}@${source.url}#${desiredVersion}`];
  } else {
    return [`${depName}@${desiredVersion}`];
  }
```

Did this in the lockfile updater: https://github.com/dependabot/dependabot-core/blob/b51a95558cd4a3dfc0b7825e964136571df304f8/npm_and_yarn/lib/dependabot/npm_and_yarn/file_updater/npm_lockfile_updater.rb#L248-L275

Wondering if it's worth extracting that to a helper and share the same logic to build install args?

Reply:
Yeah, that's a good shout 👍

Reply:
maybe a bit gnarly extracting it currently as it relies on `updated_version_requirement_for_dependency` which pulls stuff out of the `package.json`, maybe bring back the same logic we had in JS to build git install args and add a todo to fix?